src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -468,10 +468,15 @@ public sealed class JournalApplicator : IDisposable
             private readonly ManualResetEventSlim _waitForJournalStateUpdateUnderTx = new ManualResetEventSlim();
             private readonly LockTaskResponsible _flushLockTaskResponsible;
 
-            private long _lastFlushedTransactionId;
-            private long _lastFlushedJournalId;
+            public class LastFlushState
+            {
+               public long TransactionId;
+               public long JournalId;
+               public JournalFile Journal;
+            }
```

Comment:
As we never mutate those values (since `LastFlushState` is the snapshot) I'd mark all of those fields as `readonly` and create a dedicated constructor to assign them

Reply:
I'm not following?

Reply:
Can we mark entire `LastFlushState` object as `null` then instead?

Reply:
O.k. sound good

Reply:
I checked it and this leads to some problems. We lose the information about the `TransactionId` and `JournalId` and the code rely on them in `ApplyLogsToDataFile` and in the gather details for reports

Reply:
OK. Then we can leave `Journal` field as not readonly or create a dedicated boolean property (not readonly) to mark that this has been already synced.

Reply:
The journal is assigned only on `UpdateJournalStateUnderWriteTransactionLock`, with the rest
I'm not following on the issue here?

Reply:
Better to add a flag, which we can set when we finished syncing.

Reply:
I will use `new SingleUseFlag();`
O.k.?

Reply:
ye